code/modules/mechanics/MechanicMadness.dm

```diff
@@ -502,11 +501,24 @@
 	proc/componentSay(var/string)
 		string = trim(sanitize(html_encode(string)))
 		var/maptext = null
-		var/mob/user = usr
-		if (src_exists_inside_user_or_user_storage && !src.storage)
-			maptext = make_chat_maptext(src.owner, "[string]", "color: #FFBF00;", alpha = 255)
+		var/loc = null //Location used for center of all_hearers scan "Probably where you want your text attached to."
+
+		if(src.stored?.linked_item && !src.storage) //If the sound synth is attached inside something.
+			if(istype_exact(src.stored?.linked_item, /obj/item/storage/mechanics/housing_large)) //Mech Cabinet check
+				loc = src.loc
+
+			else if(istype_exact(src.stored?.linked_item, /obj/item/storage/mechanics/housing_handheld)) //Device Frame check
+				if(!obj_loc_chain(src)) //If not stored anywhere usually in your hands or on the ground
+					loc = src.loc
+
+				else //Handles rest of cases where it is stored somewhere whether on person in a container or some combination
+					var/list/atom/movable/loc_chain = obj_loc_chain(src)
+					loc = loc_chain[length(loc_chain)] //location of stop most container or possibly a mob.
```

Comment:
I feel like you couldd remove the other cases for the housings and always just call this, it should handle everything correctly unless I'm missing something. Could you try if that works?

Reply:
I'll have to check for sure on the cabinet but it does not work in some cases as it will report null so I forced a check just in case.